book/website/communication/binder/zero-to-binder.md

```diff
@@ -0,0 +1,526 @@
+
+```{note}
+Sometimes Binder's build logs prints things in red font, such as warnings that `pip` is not up-to-date (`pip` is often out of date because it's regularly updated!) or installation messages, especially if you're using R.
+These red messages don't necessarily mean there's a problem with your build and it will fail - it's just an unfortunate font colour choice!
+```
+
+(z2b-background-2)=
+### What's happening in the background? - Part 2
+
+This time, BinderHub will read the configuration file you added and install the specific version of the package you requested.
+
+(z2b-dependencies)=
+### More on pinning dependencies
+
+````{tabbed} Python
+In the above example, we used two equals signs (`==`) to pin the version of `numpy`.
+This tells Binder to install that _specific_ version.
+
+Another way to pin a version number is to use the greater than or equal to sign (`>=`) to allow any version above a particular one to be installed.
+This is useful when you have a lot of dependencies that may have dependencies on each other and allows Binder to find a configuration of your dependencies that do not conflict with one another whilst avoiding any earlier versions which may break or change your code.
+
+Finally, you could not provide a version number at all (just the name of the library/package) and Binder will install the latest version of that package.
+````
+
+````{tabbed} Julia
+In the above example, we copied a hash into our `Project.toml` file which is related to the version of the package we'd like to install.
+For a full dependency graph, we would also need to include a `Manifest.toml` file which would document dependencies of dependencies.
+Between these two files, we are able to instantiate an exact replication of a Julia environment.
+
+Of course we can imagine that, as the environment grows and the inter-dependencies become more complex, it would become very taxing to write these files by hand!
+The truth is that you'd never do it manually, the built-in package manager `Pkg` can [generate them automatically](https://julialang.github.io/Pkg.jl/v1/environments/).
+````
+
+````{tabbed} R
+In the above example, we specified that we want to use R in our project by including a date in `runtime.txt`.
+The date tells Binder which MRAN snapshot to source R and packages from.
+[MRAN](https://mran.microsoft.com/) hosts a "CRAN Time Machine" of daily snapshots of the CRAN R packages and R releases as far back as Sept. 17, 2014.
+In the above example, the MRAN snapshot dated `r-2020-05-26` is used and the version of R and `readr` available at that date and installed.
+For the workflow to work correctly, please ensure you do not supply a date earlier than this example date.
+
+This provides some rudimentary package versioning for R users but is not as robust as pinning versions in a `requirements.txt` in Python.
+For more robust and specific version pinning in R, have a look at the [`renv`](https://rstudio.github.io/renv/) package.
+````
```

Comment:
Hey @yuvipanda! Can you give some guidance on how to update this section given Binder's move to RSPM?

Reply:
Thanks for the ping, @sgibson91!

Once https://github.com/jupyterhub/repo2docker/pull/1104 lands, the only difference is that the CRAN snapshots will be maintained by RStudio and come from https://packagemanager.rstudio.com/client/#/ instead of from Microsoft / MRAN. This is *much* faster and gives us binary builds. 

Only additional point is that for some complex R packages, you will need to install some apt packages to get them to work. For example, [rstan](https://packagemanager.rstudio.com/client/#/repos/1/packages/rstan) needs `libv8-dev` to be specified in `apt.txt`. packagemanager.rstudio.com lists them explicitly in their package page, and it's not needed for most common packages - so maybe not useful for this guide? Just wanted to note it as a point of difference from current setup.

Hope this was helpful!

Reply:
Addressed in [`9e612d1`](https://github.com/alan-turing-institute/the-turing-way/pull/1850/commits/9e612d1edf2260de36bea1f6d7773f554232e53c)